hi-ml-cpath/src/health_cpath/models/deepmil.py

```diff
@@ -164,8 +165,10 @@ def log_metrics(self, stage: str) -> None:
             else:
                 log_on_epoch(self, f'{stage}/{metric_name}', metric_object)
 
-    def forward(self, instances: Tensor) -> Tuple[Tensor, Tensor]:  # type: ignore
-        should_enable_encoder_grad = torch.is_grad_enabled() and self.encoder_params.is_finetune
+    def get_instance_features(self, instances: Tensor) -> Tensor:
+        should_enable_encoder_grad = torch.is_grad_enabled() and self.encoder_params.tune_encoder
+        if not self.encoder_params.tune_encoder:
+            self.encoder.eval()
         with set_grad_enabled(should_enable_encoder_grad):
```

Comment:
Is it still necessary to call `set_grad_enabled()` here and in `get_attentions_and_bag_features()`, if the respective modules' `requires_grad` flags are already configured?

Reply:
Good point, I kept it as it was before but maybe it's unnecessary.

Reply:
I just fear it may cause confusion, and any changes to the fine-tuning would need to be replicated in multiple places.